crates/lib/src/executable.rs

```diff
@@ -299,14 +299,15 @@ impl<'executable> Executable<'executable, '_> {
         self
     }
 
-    /// Load `self.client` if not yet loaded, then return a reference to it.
-    async fn get_qcs_client(&mut self) -> Result<Arc<Qcs>, Error> {
+    /// Get a reference to the [`Qcs`] client used by the executable,
+    /// which loads and stores a default client if one has not been set.
+    pub async fn get_qcs_client(&mut self) -> Arc<Qcs> {
```

Comment:
Since we are making this into a public method, we may want to consider more closely following the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter) for naming such things. In this case, I think the advice would be to just call this `qcs_client()`.